distribution/lib/Standard/Base/0.0.0-dev/src/Data/Vector.enso

```diff
@@ -1563,3 +1565,24 @@ type No_Wrap
 
 ## PRIVATE
 Wrapped_Error.from (that : Map_Error) = Wrapped_Error.Value that that.inner_error
+
+type Vector_Comparator
+    compare x y =
+        min_length = x.length.min y.length
+        when_prefixes_equal =
+            ## At this point, if the vectors are the same length, then they are
+               identical; otherwise, the shorter one is lesser.
+            if x.length == y.length then Ordering.Equal else
+                Ordering.compare x.length y.length
+        go i =
+            if i >= min_length then when_prefixes_equal else
+                x_elem = x.at i
+                y_elem = y.at i
+                Ordering.compare x_elem y_elem . and_then <|
+                    @Tail_Call go (i + 1)
+        k = go 0
+        k
+
+    hash x = Default_Comparator.hash_builtin x
+
+Comparable.from (that : Vector) = Comparable.new that Vector_Comparator
```

Comment:
I'm pretty sure this just re-implements `Vector_Lexicographic_Order.compare`?

I think we did not register it into `Comparable` on purpose - comparing vectors is not unambiguously defined operation - there's multiple orderings that may seem right depending on context. So we did not want to provide any default one to avoid user confusion and require to choose one explicitly (we implement only lexicographic one but for example an (incomplate) ordering could be the point-wise order.

Of course we can decide that it is beneficial to include a default `<` for vectors. But I think the original decision was not wrong.

Reply:
The downside to making the comparable is that if a user accidentally nests vectors, and sorts them, they won't know they made a mistake. But it will be easy to see that the data is nested. Without a comparator, they'll get an error.

The upside is that they'll be able to sort vectors containing more things.

@JaroslavTulach You asked if modern languages allow this; Haskell does:
```
 main = do
   msp $ [1, 2] < [3, 4]
=> True
```